minimal-plugin/src/main/java/io/micronaut/gradle/PluginsHelper.java

```diff
@@ -126,4 +126,19 @@ public static MicronautRuntime resolveRuntime(Project p) {
         }
         return micronautRuntime;
     }
+
+    public static ApplicationType resolveApplicationType(Project p) {
+        MicronautExtension ext = p.getExtensions().findByType(MicronautExtension.class);
+        Object o = p.findProperty("micronaut.applicationType");
```

Comment:
:x: this is not how it should be done. Instead, the convention value for `applicationType` should be set to `providers.gradleProperty("micronaut.applicationType")`

Reply:
Please, note that this method is identical to `PluginsHelper::resolveRuntime`. How can I obtain a `providers` instance?

Reply:
Yes, but I think new stuff should be implemented properly. Most of the plugin is quite old and uses outdated patterns. You can get the `Providers` instance from the `Project#getProviders()`

Reply:
Changed it here: https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/424/commits/8887653238e5ad6149577779673d09aff827db35

Let me know if it matches your expectations.